osu.Game.Rulesets.Mania/Mods/ManiaModPerfect.cs

```diff
@@ -1,26 +1,34 @@
-﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
-// See the LICENCE file in the repository root for full licence text.
-
-using osu.Game.Rulesets.Judgements;
-using osu.Game.Rulesets.Mods;
-using osu.Game.Rulesets.Scoring;
-
-namespace osu.Game.Rulesets.Mania.Mods
-{
-    public class ManiaModPerfect : ModPerfect
-    {
-        protected override bool FailCondition(HealthProcessor healthProcessor, JudgementResult result)
-        {
-            if (!isRelevantResult(result.Judgement.MinResult) && !isRelevantResult(result.Judgement.MaxResult) && !isRelevantResult(result.Type))
-                return false;
-
-            // Mania allows imperfect "Great" hits without failing.
-            if (result.Judgement.MaxResult == HitResult.Perfect)
-                return result.Type < HitResult.Great;
-
-            return result.Type != result.Judgement.MaxResult;
-        }
-
-        private bool isRelevantResult(HitResult result) => result.AffectsAccuracy() || result.AffectsCombo();
-    }
-}
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+
+namespace osu.Game.Rulesets.Mania.Mods
+{
+    public class ManiaModPerfect : ModPerfect
+    {
+        [SettingSource("Full perfect", "Placeholder")]
+        public BindableBool AllPerfect { get; } = new BindableBool(false);
+
+        protected override bool FailCondition(HealthProcessor healthProcessor, JudgementResult result)
+        {
+            if (!isRelevantResult(result.Judgement.MinResult) && !isRelevantResult(result.Judgement.MaxResult) && !isRelevantResult(result.Type))
+                return false;
+
+            if (AllPerfect.Value)

```

Comment:
Pretty sure this needs a `Judgement.MaxResult == Perfect` check like the branch just below. Unless failing on a hold break was your implicit intention or something.

In fact you could probably combine the two conditions at that point.